build/ci/github_actions/operator/wait_for_driver_ready.sh

```diff
@@ -0,0 +1,69 @@
+#!/bin/bash -xe
+set +o pipefail
+
+driver_is_ready=false
+actual_driver_running_time_in_seconds=0
+minimum_driver_running_time_in_seconds=10
+declare -a driver_pods_types=(
+  "controller"
+  "node"
+)
+
+get_csi_pods (){
+  kubectl get pod -A -l csi
+}
+
+wait_for_driver_pod_to_start (){
+  driver_pod_type=$1
+  while [ "$(get_csi_pods | grep $driver_pod_type | wc -l)" -eq 0 ]; do
+    echo "The $driver_pod_type is not deployed"
+    sleep 1
+  done
+}
+
+wait_for_driver_deployment_to_start (){
+  for driver_pods_type in "${driver_pods_types[@]}"
+  do
+    wait_for_driver_pod_to_start $driver_pods_type
+  done
+}
+
+wait_for_driver_deployment_to_finish (){
+  while [ $driver_is_ready == "false" ]; do
+    if [ "$(get_csi_pods | grep -iv running | grep -iv name | wc -l)" -eq 0 ]; then
+      ((++actual_driver_running_time_in_seconds))
+      if [ $actual_driver_running_time_in_seconds -eq $minimum_driver_running_time_in_seconds ]; then
+        driver_is_ready=true
+      fi
+    else
+      actual_driver_running_time_in_seconds=0
+    fi
+    get_csi_pods
+    sleep 1
+  done
+}
+
+is_wanted_image_in_pod (){
+  pod_type=$1
+  wanted_pod_image=$2
+  image_in_pod=`kubectl describe pod $(get_csi_pods | grep $pod_type) | -i image: | grep $wanted_pod_image | awk -F Image: '{print$2}' | awk '{print$1}'`
+  if [[ $image_in_pod != $wanted_pod_image ]]; then
```

Comment:
do we grep for the pod image and then make sure its there?

Reply:
how can it not be there if it passed the grep?

Reply:
I wanted that we could control the output message in the fail that we will understand that it failed on wrong image

Reply:
not sure I completely follow, could you write here a theoretical output for which the condition of the `if` would be false?

Reply:
if the `if` will be false the script will give [this](https://github.com/IBM/ibm-block-csi-operator/blob/df5f840e897bd3934b232211c091d25038c08cff/build/ci/github_actions/operator/wait_for_driver_ready.sh#L51) output, but if the script will fail in the grep it won't show any errors

Reply:
I meant a `kubectl describe pod` output, not this script's output

Reply:
done